### PR TITLE
add claims for submariner

### PR DESCRIPTION
**Pull request by zhiweiyin318**
task: https://github.com/open-cluster-management/backlog/issues/8114
add 2 claims:
1. region.open-cluster-management.io,  the region for OCP on AWS and GCP.
2. infrastructure.openshift.io , the infrastructure config of OCP, format {"infraName":"xxx"}

### CODE REVIEW

File: pkg/klusterlet/clusterinfo/clusterinfo_controller.go [hunk 1]
```diff
@@ -597,6 +667,12 @@ func (r *ClusterInfoReconciler) GetClusterClaims() ([]*clusterv1alpha1.ClusterCl
 	if len(distributionInfo.OCP.Version) > 0 {
 		claims = append(claims, newClusterClaim("version.openshift.io", distributionInfo.OCP.Version))
 	}
+
+	infraName := r.getInfraName()
+	if len(infraName) > 0 {
+		claims = append(claims, newClusterClaim("infrastructure.openshift.io", infraName))
```
Comment (skeeey) [inline pkg/klusterlet/clusterinfo/clusterinfo_controller.go:673]: for infrastructure, we may use a json string, e.g. `{"infraName":"cluster-1234"}`

Comment (zhiweiyin318) [reply pkg/klusterlet/clusterinfo/clusterinfo_controller.go]: done
```
- name: infrastructure.openshift.io
    value: '{"infraName":"yzw-8tlfw"}'
- name: region.open-cluster-management.io
    value: us-east-1
```

File: pkg/klusterlet/clusterinfo/clusterinfo_controller.go [hunk 2]
```diff
@@ -567,6 +567,77 @@ func (r *ClusterInfoReconciler) getDistributionInfoAndClusterID() (clusterv1beta
...
+		return ""
+	}
+
+	infraName, _, err := unstructured.NestedString(obj.Object, "status", "infrastructureName")
+	if err != nil {
+		klog.Errorf("failed to get OCP infrastructure Name in status of infrastructures.config.openshift.io/cluster: %v", err)
+		return ""
+	}
+
+	infraConfig := fmt.Sprintf("{\"infraName\":\"%s\"}", infraName)
```
Comment (qiujian16) [inline pkg/klusterlet/clusterinfo/clusterinfo_controller.go:637]: it is better to have a struct for this and use json.Marshal

Comment (zhiweiyin318) [reply pkg/klusterlet/clusterinfo/clusterinfo_controller.go]: done
```
type InfraConfig struct {
	InfraName string `json:"infraName,omitempty"`
}
```

**Pull request merged**